base/REPLCompletions.jl

```diff
@@ -130,8 +130,12 @@ function complete_path(path::AbstractString, pos)
             push!(matches, id ? file * (@windows? "\\\\" : "/") : file)
         end
     end
-    matches = UTF8String[replace(s, r"\s", "\\ ") for s in matches]
-    return matches, nextind(path, pos - sizeof(prefix) - length(matchall(r" ", prefix))):pos, length(matches) > 0
+    matches = UTF8String[replace(s, r" ", "\\ ") for s in matches]
```

Comment:
yes, this lines change is actually not for fixing the bug. The reason for this is to replace space in path with `"\ "` so it is correctly escaped for paths and files. I have incidentally introduced `"\s"` instead of `" "` in pull #8838. It do not seem to do any difference since file names cannot contain tabs, return carriage and new line. However I think it is more correct to only match `" "`